server/runtime/src/main/java/org/infinispan/server/Server.java

```diff
@@ -508,7 +509,7 @@ public synchronized CompletableFuture<ExitStatus> run() {
                   singlePortRouter.ssl().enabled() ? "https" : "http", singlePortRouter.host(), singlePortRouter.port()
             );
          }
-         cacheManagerStart.whenComplete((ignore, t) -> {
+         initialization = cacheManagerStart.whenComplete((ignore, t) -> {
```

Comment:
This seems unneeded. Can we not just widen the try/catch inside the `whenComplete` lambda that catches Throwable which just invokes `r.completeExceptionally(t);` ? And maybe change the try/catch of the sync code path to be Throwable as well.